### PR TITLE
✨ feat: migrate less to token for Avatar

### PR DESCRIPTION
**Pull request by heiyu4585**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md?plain=1)]

### 🤔 This is a ...

- [x] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Workflow
- [ ] Other (about what?)

### 💡 Background and solution
https://github.com/ant-design/ant-design/issues/41884

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |     migrate less to token for Avatar      |
| 🇨🇳 Chinese |     迁移 `Avatar`  组件的 `less token`       |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

---

### <samp>🤖 Generated by Copilot at cf64ed2</samp>

This pull request refactors the style of the `Avatar` component to use component tokens and updates the documentation for migrating less variables to component tokens in both English and Chinese versions. It also fixes a typo in the document title.

### <samp>🤖 Generated by Copilot at cf64ed2</samp>

* Refactor the `genComponentStyleHook` function to accept a third argument that returns the component token properties based on the global token ([link](https://github.com/ant-design/ant-design/pull/42063/files?diff=unified&w=0#diff-9bae49959a354c48caf848def36049b3c0d1e929e594370e57fc1a40c8a10f38L130-R166))
* Move the `ComponentToken` interface to the top of the file and add the `avatarBg` and `avatarColor` properties to it ([link](https://github.com/ant-design/ant-design/pull/42063/files?diff=unified&w=0#diff-9bae49959a354c48caf848def36049b3c0d1e929e594370e57fc1a40c8a10f38L2-R6))
* Remove the `avatarGroupOverlapping` and `avatarGroupSpace` properties from the `AvatarToken` type as they are deprecated ([link](https://github.com/ant-design/ant-design/pull/42063/files?diff=unified&w=0#diff-9bae49959a354c48caf848def36049b3c0d1e929e594370e57fc1a40c8a10f38L17-R20))
* Document the mapping between the less variables and the component token properties for the `Avatar` component and the deprecated variables in `docs/react/migrate-less-variables.en-US.md` and `docs/react/migrate-less-variables.zh-CN.md` ([link](https://github.com/ant-design/ant-design/pull/42063/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aR39-R56), [link](https://github.com/ant-design/ant-design/pull/42063/files?diff=unified&w=0#diff-dff3e0a0a5fae848621e55da4e2adaf0ec6c57d5e29d3f601799c8898c280739R39-R56))
* Fix a typo in the title of the `Modal` section in `docs/react/migrate-less-variables.en-US.md` ([link](https://github.com/ant-design/ant-design/pull/42063/files?diff=unified&w=0#diff-999566baa64628c49813370e7cb13c3c8f5e0c99ef8fe9cec7c5052985c10d9aL10-R10))

### CODE REVIEW

File: docs/react/migrate-less-variables.en-US.md [hunk 1]
```diff
@@ -17,6 +17,24 @@ This document contains the correspondence between all the less variables related
...
+| `@avatar-size-sm` | `avatarSizeSM` | - |
+| `@avatar-font-size-base` | `avatarFontSizeBase` | - |
+| `@avatar-font-size-lg` | `avatarFontSizeLG` | - |
+| `@avatar-font-size-sm` | `avatarFontSizeSM` | - |
+| `@avatar-bg` | `avatarBg` | - |
+| `@avatar-color` | `avatarColor` | - |
+| `@avatar-border-radius` | - | Deprecated for style change |
+| `@avatar-group-overlapping` | - | Deprecated for style change |
+| `@avatar-group-space` | - | Deprecated for style change |
+| `@avatar-group-border-color` | - | Deprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:36]: 这两个有的，文档可以补一下

File: docs/react/migrate-less-variables.en-US.md [hunk 2]
```diff
@@ -17,6 +17,24 @@ This document contains the correspondence between all the less variables related
...
+| --- | --- | --- |
+| `@avatar-size-base` | `avatarSizeBase` | - |
+| `@avatar-size-lg` | `avatarSizeLG` | - |
+| `@avatar-size-sm` | `avatarSizeSM` | - |
+| `@avatar-font-size-base` | `avatarFontSizeBase` | - |
+| `@avatar-font-size-lg` | `avatarFontSizeLG` | - |
+| `@avatar-font-size-sm` | `avatarFontSizeSM` | - |
+| `@avatar-bg` | `avatarBg` | - |
+| `@avatar-color` | `avatarColor` | - |
+| `@avatar-border-radius` | - | Deprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:33]: 可以找个对应的 GlobalToken 写上去

File: docs/react/migrate-less-variables.en-US.md [hunk 3]
```diff
@@ -17,6 +17,24 @@ This document contains the correspondence between all the less variables related
 
 <!-- ### Avatar -->
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:18]:
```suggestion
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 4]
```diff
@@ -17,6 +17,24 @@ title: Less 变量迁移 Design Token
 
 <!-- ### Avatar 头像 -->
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:18]:
```suggestion
```

File: docs/react/migrate-less-variables.en-US.md (hunk 3 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:18]: done

File: docs/react/migrate-less-variables.zh-CN.md (hunk 4 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.zh-CN.md:18]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 1 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:36]: done

File: components/avatar/style/index.ts [hunk 5]
```diff
@@ -14,11 +12,11 @@ type AvatarToken = FullToken<'Avatar'> & {
   avatarFontSizeBase: number;
   avatarFontSizeLG: number;
   avatarFontSizeSM: number;
-  avatarGroupOverlapping: number;
```
[the remark this answers is not in the document]
Comment (heiyu4585) [reply components/avatar/style/index.ts:17]: done

File: docs/react/migrate-less-variables.en-US.md (hunk 2 above)
Comment (heiyu4585) [reply docs/react/migrate-less-variables.en-US.md:31]: done

File: docs/react/migrate-less-variables.en-US.md [hunk 6]
```diff
@@ -15,7 +15,23 @@ This document contains the correspondence between all the less variables related
...
+| --- | --- | --- |
+| `@avatar-size-base` | `avatarSizeBase` | - |
+| `@avatar-size-lg` | `avatarSizeLG` | - |
+| `@avatar-size-sm` | `avatarSizeSM` | - |
+| `@avatar-font-size-base` | `avatarFontSizeBase` | - |
+| `@avatar-font-size-lg` | `avatarFontSizeLG` | - |
+| `@avatar-font-size-sm` | `avatarFontSizeSM` | - |
+| `@avatar-bg` | `avatarBg` | - |
+| `@avatar-color` | `avatarColor` | - |
+| `@avatar-border-radius` | - | Deprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:31]:
```suggestion
| `@avatar-border-radius` | `borderRadius` | Global Token |
```

File: docs/react/migrate-less-variables.en-US.md [hunk 7]
```diff
@@ -15,7 +15,23 @@ This document contains the correspondence between all the less variables related
...
+| `@avatar-group-overlapping` | - | Deprecated for style change |
+| `@avatar-group-space` | `groupSpace` | - |
+| `@avatar-group-border-color` | `groupBorderColor` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:34]:
```suggestion
| `@avatar-group-border-color` | `colorBorderBg` | Global Token |
```

File: docs/react/migrate-less-variables.en-US.md [hunk 8]
```diff
@@ -15,7 +15,23 @@ This document contains the correspondence between all the less variables related
...
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@avatar-size-base` | `avatarSizeBase` | - |
+| `@avatar-size-lg` | `avatarSizeLG` | - |
+| `@avatar-size-sm` | `avatarSizeSM` | - |
+| `@avatar-font-size-base` | `avatarFontSizeBase` | - |
+| `@avatar-font-size-lg` | `avatarFontSizeLG` | - |
+| `@avatar-font-size-sm` | `avatarFontSizeSM` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:28]:
```suggestion
| `@avatar-font-size-base` | `textFontSize` | - |
| `@avatar-font-size-lg` | `textFontSizeLG` | - |
| `@avatar-font-size-sm` | `textFontSizeSM` | - |
```

File: docs/react/migrate-less-variables.en-US.md [hunk 9]
```diff
@@ -15,7 +15,23 @@ This document contains the correspondence between all the less variables related
...
+| `@avatar-bg` | `avatarBg` | - |
+| `@avatar-color` | `avatarColor` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:30]:
```suggestion
| `@avatar-color` | `colorTextLightSolid` | Global Token |
```

File: components/avatar/style/index.ts [hunk 10]
```diff
@@ -110,55 +112,60 @@ const genBaseStyle: GenerateStyle<AvatarToken> = (token) => {
...
       display: 'inline-flex',
 
       [`${componentCls}`]: {
-        borderColor: avatarGroupBorderColor,
+        borderColor: groupBorderColor,
       },
 
       [`> *:not(:first-child)`]: {
-        marginInlineStart: avatarGroupSpace,
+        marginInlineStart: groupSpace,
```
Comment (MadCcc) [inline components/avatar/style/index.ts:126]: 这个应该是 groupOverlapping

File: docs/react/migrate-less-variables.en-US.md [hunk 11]
```diff
@@ -15,7 +15,23 @@ This document contains the correspondence between all the less variables related
...
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@avatar-size-base` | `avatarSizeBase` | - |
+| `@avatar-size-lg` | `avatarSizeLG` | - |
+| `@avatar-size-sm` | `avatarSizeSM` | - |
+| `@avatar-font-size-base` | `avatarFontSizeBase` | - |
+| `@avatar-font-size-lg` | `avatarFontSizeLG` | - |
+| `@avatar-font-size-sm` | `avatarFontSizeSM` | - |
+| `@avatar-bg` | `avatarBg` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.en-US.md:29]:
```suggestion
| `@avatar-bg` | - | 可由 `className` 或 `style` 直接覆盖 |
```

File: docs/react/migrate-less-variables.zh-CN.md [hunk 12]
```diff
@@ -15,7 +15,23 @@ title: Less 变量迁移 Design Token
...
+| `@avatar-size-lg` | `sizeLG` | - |
+| `@avatar-size-sm` | `sizeSM` | - |
+| `@avatar-font-size-base` | `textFontSize` | - |
+| `@avatar-font-size-lg` | `textFontSizeLG` | - |
+| `@avatar-font-size-sm` | `textFontSizeSM` | - |
+| `@avatar-bg` | - | 可由 `className` 或 `style` 直接覆盖 |
+| `@avatar-color` | `colorTextLightSolid` | 全局 Token |
+| `@avatar-border-radius` | `borderRadius` | 全局 Token |
+| `@avatar-group-overlapping` | `groupOverlapping` | - |
+| `@avatar-group-space` | `groupSpace` | Deprecated for style change |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:33]: 这个不应该废弃的，可以对比一下 v4 和 v5，这个会用在 popover 里的 avatar，v5 现在应该是有问题的

Comment (heiyu4585) [reply docs/react/migrate-less-variables.zh-CN.md]: done

File: docs/react/migrate-less-variables.zh-CN.md [hunk 13]
```diff
@@ -15,7 +15,23 @@ title: Less 变量迁移 Design Token
...
 
-<!-- ### Avatar 头像 -->
+### Avatar 头像
+
+<!-- prettier-ignore -->
+| less 变量 | Component Token | 备注 |
+| --- | --- | --- |
+| `@avatar-size-base` | `size` | - |
+| `@avatar-size-lg` | `sizeLG` | - |
+| `@avatar-size-sm` | `sizeSM` | - |
```
Comment (MadCcc) [inline docs/react/migrate-less-variables.zh-CN.md:25]:
```suggestion
| `@avatar-size-base` | `containerSize` | - |
| `@avatar-size-lg` | `containerSizeLG` | - |
| `@avatar-size-sm` | `containerSizeSM` | - |
```

File: components/avatar/style/index.ts [hunk 14]
```diff
@@ -110,55 +111,67 @@ const genBaseStyle: GenerateStyle<AvatarToken> = (token) => {
...
-    avatarColor: colorTextLightSolid,
-
-    avatarSizeBase: controlHeight,
-    avatarSizeLG: controlHeightLG,
-    avatarSizeSM: controlHeightSM,
+export default genComponentStyleHook(
+  'Avatar',
+  (token) => {
+    const avatarToken = mergeToken<AvatarToken>(token, {});
+    console.log([genBaseStyle(avatarToken), genGroupStyle(avatarToken)]);
```
Comment (MadCcc) [inline components/avatar/style/index.ts:140]:
```suggestion
```

File: components/avatar/style/index.ts [hunk 15]
```diff
@@ -110,55 +111,67 @@ const genBaseStyle: GenerateStyle<AvatarToken> = (token) => {
...
+      containerSize: controlHeight,
+      containerSizeLG: controlHeightLG,
+      containerSizeSM: controlHeightSM,
+
+      fontSize: Math.round((fontSizeLG + fontSizeXL) / 2),
+      fontSizeLG: fontSizeHeading3,
+      fontSizeSM: fontSize,
+
+      avatarBg: colorTextPlaceholder,
+      avatarColor: colorTextLightSolid,
```
Comment (MadCcc) [inline components/avatar/style/index.ts:171]: 这些都不用在 组件 token 处赋值，都可以还原回去

File: components/avatar/style/index.ts [hunk 16]
```diff
@@ -1,22 +1,24 @@
...
+  fontSize: number;
+  fontSizeLG: number;
+  fontSizeSM: number;
+  groupSpace: number;
+  groupOverlapping: number;
+  groupBorderColor: string;
+}
+
+type AvatarToken = FullToken<'Avatar'> & {
   avatarGroupOverlapping: number;
```
Comment (MadCcc) [inline components/avatar/style/index.ts:21]: 这个应该已经没用了吧

File: components/avatar/style/index.ts [hunk 17]
```diff
@@ -1,22 +1,24 @@
...
 import type { FullToken, GenerateStyle } from '../../theme/internal';
 import { genComponentStyleHook, mergeToken } from '../../theme/internal';
-import { resetComponent } from '../../style';
-
-export interface ComponentToken {}
 
-type AvatarToken = FullToken<'Avatar'> & {
+export interface ComponentToken {
   avatarBg: string;
   avatarColor: string;
```
Comment (MadCcc) [inline components/avatar/style/index.ts:8]: 这两个不是组件 token

File: components/avatar/style/index.ts (hunk 16 above)
Comment (heiyu4585) [reply components/avatar/style/index.ts:21]: done

File: components/avatar/style/index.ts (hunk 17 above)
Comment (heiyu4585) [reply components/avatar/style/index.ts]: done

File: components/avatar/style/index.ts [hunk 18]
```diff
@@ -1,23 +1,24 @@
...
-import { resetComponent } from '../../style';
 
-export interface ComponentToken {}
+export interface ComponentToken {
+  containerSize: number;
+  containerSizeLG: number;
+  containerSizeSM: number;
+  fontSize: number;
+  fontSizeLG: number;
+  fontSizeSM: number;
```
Comment (MadCcc) [inline components/avatar/style/index.ts:12]:
```suggestion
  textFontSize: number;
  textFontSizeLG: number;
  textFontSizeSM: number;
```
和文档里的统一

Comment (heiyu4585) [reply components/avatar/style/index.ts:12]: done

File: components/avatar/demo/component-token.tsx [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+import React from 'react';
+
+const App: React.FC = () => (
+  <ConfigProvider
+    theme={{
+      components: {
+        Avatar: {
+          size: 60,
+          sizeLG: 30,
+          sizeSM: 16,
```
Comment (MadCcc) [inline components/avatar/demo/component-token.tsx:12]:
```suggestion
          containerSize: 60,
          containerSizeLG: 30,
          containerSizeSM: 16,
```

**Review by MadCcc — APPROVED** [review]

**Pull request merged**